opacus/privacy_engine.py

```diff
@@ -350,15 +350,22 @@ def dp_step(self, closure=None, is_empty=False):
             if hasattr(self.privacy_engine.module, "ddp_hooks"):
                 # We just update the accountant
                 self.privacy_engine.steps += 1
-
+                return self.original_step(closure)
             else:
+                loss = None
+                if closure is not None:
+                    with torch.enable_grad():
+                        loss = closure()
+
                 self.privacy_engine.step(is_empty)
                 if isinstance(
                     self.privacy_engine.module._module,
                     DifferentiallyPrivateDistributedDataParallel,
                 ):
                     average_gradients(self.privacy_engine.module)
-            self.original_step(closure)
+
+                self.original_step()
```

Comment:
Hmm, I'm not sure if I should pass the closure to the original_step. 
- If I do, it probably calls it again and we run the backward pass twice. 
- If I do NOT, the optimizers requiring several model reevaluations must not work. Although don't know how that case affects grad sampling and accounting.

@alexandresablayrolles maybe you know the answer?